### PR TITLE
Correcting exported cmake components for tl-expected/all [help wanted]

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **tl-expected/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

modifying test package to use exported targets. this current fails. 

blocks #1649

### CODE REVIEW

File: recipes/tl-expected/all/conanfile.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def package_id(self):
         self.info.header_only()
 
     def package_info(self):
+        self.cpp_info.name = "tl"
         self.cpp_info.names["cmake_find_package"] = "tl-expected"
         self.cpp_info.names["cmake_find_package_multi"] = "tl-expected"
         self.cpp_info.components["expected"].name = "expected"
```
Comment (prince-chrismc) [inline recipes/tl-expected/all/conanfile.py:43]: I am not sure the syntax is expressive enough to I can match what it output by this project.

File: recipes/tl-expected/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,8 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.1)
 project(PackageTest CXX)
 set(CMAKE_CXX_STANDARD 11)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+find_package(tl-expected REQUIRED)
```
Comment (prince-chrismc) [inline recipes/tl-expected/all/test_package/CMakeLists.txt:5]: https://github.com/cpp-netlib/url/blob/0f33d2a242ea79feb60e662b227e1cff37ea394f/CMakeLists.txt#L34

File: recipes/tl-expected/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -1,9 +1,8 @@
...
 
 add_executable(example example.cpp)
-target_link_libraries(example ${CONAN_LIBS})
+target_link_libraries(example tl::expected)
```
Comment (prince-chrismc) [inline recipes/tl-expected/all/test_package/CMakeLists.txt:8]: https://github.com/cpp-netlib/url/blob/0f33d2a242ea79feb60e662b227e1cff37ea394f/src/CMakeLists.txt#L138-L142

**Comment by prince-chrismc** [comment]
Same as #2003

File: recipes/tl-expected/all/test_package/conanfile.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 class ExpectedTestConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "cmake_find_package"
```
Comment (madebr) [inline recipes/tl-expected/all/test_package/conanfile.py:6]: Please always keep `cmake` in the generators of `test_package`.
The `cmake` generator ensures that the compiler is configured according to the profile. 
```suggestion
    generators = "cmake", "cmake_find_package"
```

File: recipes/tl-expected/all/test_package/CMakeLists.txt (hunk 2 above)
Comment (madebr) [inline recipes/tl-expected/all/test_package/CMakeLists.txt:5]:
```suggestion
include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
conan_basic_setup(TARGETS)
find_package(tl-expected REQUIRED)
```

File: recipes/tl-expected/all/conanfile.py [hunk 5]
```diff
@@ -37,6 +37,7 @@ def package_id(self):
         self.info.header_only()
 
     def package_info(self):
+        self.cpp_info.name = "tl"
```
Comment (prince-chrismc) [inline recipes/tl-expected/all/conanfile.py:40]:
```suggestion
```

**Comment by prince-chrismc** [comment]
```
-- Conan: Using autogenerated Findtl-expected.cmake
-- Found tl-expected: 1.0.0 (found version "1.0.0") 
-- Configuring done
CMake Error at CMakeLists.txt:9 (add_executable):
  Target "example" links to target "tl::expected" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```

Seems like components are not expressive enough :cry:

**Pull request closed without merging**